### DISCUSSION TITLE
How to filter/hide private files from beeing included?

### DISCUSSION THREAD

**Discussion by braniii** [mkdocstrings/mkdocstrings#530] [Q&A] [answered]
Hi there,

I am using mkdocstrings to automatically generate my documentation and I have to say that this package is doing a great job! My problem is, that I have modules with many methods which brought me to the idea to split my module (here called `b`) into two submodules which are independent and some general functions into the submodule `_b` which are imported into the `b.__init__.py` to avoid lengthly init files. Using `__all__` in the `b.__init__.py` I managed to show the methods in the documentation of `b`.

```
src/mypackage
├── __init__.py
├── a.py
└── b
    ├── __init__.py
    ├── _b.py
    ├── b1.py
    └── b2.py
```

And here my question. How do I prevent mkdocstrings from creating the documentation of `_b.py`? Similar to the option `filters` option I would like to filter also by file names, e.g. files with leading underscores.
I am using mkdocstrings with the python(experimental) handler.

**Comment by pawamoy** [comment, marked as answer]
Hello! How do you inject docs in your markdown pages? Do you inject the whole package docs recursively, or use the recipe from our docs to generate one page per module?

**Reply by braniii** [reply]
Hi thx for this fast reply,

I am using the [gen_ref_nav.py](https://github.com/moldyn/msmhelper/blob/main/docs/gen_ref_nav.py) based on the one provided here in the documentation.  And taking a look at it, I found the answer, including
```python
...
parts = list(module_path.parts) 
if parts[-1] == '__init__':
    ...
elif parts[-1].startswith('_'): 
    continue
```
the elif condition leads to the desired effect. Thx for pointing me to the right place to find a solution :)
